pkg/config/schema/schema.go

```diff
@@ -0,0 +1,607 @@
+	VirtualScheduler *Enablement                  `yaml:"virtualScheduler,omitempty" json:"virtualScheduler,omitempty" product:"pro"`
+	Proxy            *ControlPlaneProxy           `yaml:"proxy,omitempty" json:"proxy,omitempty"`
+	Service          ControlPlaneService          `yaml:"service,omitempty" json:"service,omitempty"`
+	Ingress          ControlPlaneIngress          `yaml:"ingress,omitempty" json:"ingress,omitempty"`
+	HighAvailability ControlPlaneHighAvailability `yaml:"highAvailability,omitempty" json:"highAvailability,omitempty"`
+	Advanced         ControlPlaneAdvanced         `yaml:"advanced,omitempty" json:"advanced,omitempty"`
+}
+
+type Distributions struct {
+	K3S      *DistributionsK3s    `yaml:"k3s,omitempty" json:"k3s,omitempty"`
+	K8S      *DistributionsK8s    `yaml:"k8s,omitempty" json:"k8s,omitempty"`
+	K0S      *corev1.Container    `yaml:"k0s,omitempty" json:"k0s,omitempty"`
+	Advanced DistributionAdvanced `yaml:"advanced,omitempty" json:"advanced,omitempty"`
+}
+
+type DistributionsK3s struct {
+	corev1.Container `yaml:",inline" json:",inline"`
+	Token            string `yaml:"token,omitempty" json:"token,omitempty"`
+}
+
+func (d *DistributionsK3s) SetDefaults() {
+	if defaults.CanUpdate(d.Container.Resources) {
+		d.Container.Resources = corev1.ResourceRequirements{
+			Limits: map[corev1.ResourceName]resource.Quantity{
+				"cpu":    {Format: "100m"},
+				"memory": {Format: "256Mi"},
+			},
+			Requests: map[corev1.ResourceName]resource.Quantity{
+				"cpu":    {Format: "100m"},
+				"memory": {Format: "256Mi"},
+			},
+		}
+	}
+}
+
+type DistributionsK8s struct {
+	APIServer         *corev1.Container `yaml:"apiServer,omitempty" json:"apiServer,omitempty"`
+	ControllerManager *corev1.Container `yaml:"controllerManager,omitempty" json:"controllerManager,omitempty"`
+	Scheduler         *corev1.Container `yaml:"scheduler,omitempty" json:"scheduler,omitempty"`
+	ETCD              *corev1.Container `yaml:"etcd,omitempty" json:"etcd,omitempty"`
+}
+
+type DistributionAdvanced struct {
+	Paths DistributionPaths `yaml:"paths,omitempty" json:"paths,omitempty"`
+}
+type DistributionPaths struct {
+	KubeConfig          string `yaml:"kubeConfig,omitempty" json:"kubeConfig,omitempty" default:"/data/server/cred/admin.kubeconfig"`
+	ServerCAKey         string `yaml:"serverCAKey,omitempty" json:"serverCAKey,omitempty" default:"/data/server/tls/server-ca.key"`
+	ServerCACert        string `yaml:"serverCACert,omitempty" json:"serverCACert,omitempty" default:"/data/server/tls/server-ca.crt"`
+	ClientCACert        string `yaml:"clientCACert,omitempty" json:"clientCACert,omitempty" default:"/data/server/tls/client-ca.crt"`
+	RequestHeaderCACert string `yaml:"requestHeaderCACert,omitempty" json:"requestHeaderCACert,omitempty" default:"/data/server/tls/request-header-ca.crt"`
+}
+
+type BackingStore struct {
+	SQLite       Enablement    `yaml:"sqlite,omitempty" json:"sqlite,omitempty"`
```

Comment:
this shouldn't be defined, since it's either the default for k3s, either not used